client/comms/wsconn_test.go

```diff
@@ -282,11 +286,16 @@ func TestWsConn(t *testing.T) {
 	}
 	waiter := dex.NewConnectionMaster(wsc)
 	err = waiter.Connect(ctx)
-	t.Log("Connect:", err)
+	if err != nil {
+		t.Fatalf("Connect: %v", err)
+	}
 
 	reconnectAndPing := func() {
-		// Drop the connection and force a reconnect by waiting.
-		time.Sleep(pingWait * 2)
+		// Drop the connection and force a reconnect by waiting longer than the
+		// read deadline (the ping wait), plus a bit extra to allow the timeout
+		// to flip off the connection and queue a reconnect.
+		time.Sleep(pingWait * 3 / 2)
+		runtime.Gosched()
```

Comment:
It's still possible for this to be insufficient, but it is at least now actually reconnecting.